client/components/data/query-jetpack-settings/index.jsx

```diff
@@ -1,44 +1,28 @@
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestJetpackSettings } from 'calypso/state/jetpack/settings/actions';
 import isRequestingJetpackSettings from 'calypso/state/selectors/is-requesting-jetpack-settings';
 
-class QueryJetpackSettings extends Component {
-	static propTypes = {
-		query: PropTypes.object,
-		siteId: PropTypes.number,
-		// Connected props
-		requestingSettings: PropTypes.bool,
-		requestJetpackSettings: PropTypes.func,
-	};
-
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId !== nextProps.siteId ) {
-			this.request( nextProps );
-		}
+const request = ( siteId, query ) => ( dispatch, getState ) => {
+	if ( ! isRequestingJetpackSettings( getState(), siteId ) ) {
+		dispatch( requestJetpackSettings( siteId, query ) );
 	}
+};
 
-	request( props ) {
-		if ( props.requestingSettings || ! props.siteId ) {
-			return;
-		}
+function QueryJetpackSettings( { siteId, query } ) {
+	const dispatch = useDispatch();
 
-		props.requestJetpackSettings( props.siteId, props.query );
-	}
+	useEffect( () => {
+		dispatch( request( siteId, query ) );
+	}, [ dispatch, siteId, query ] );
 
-	render() {
-		return null;
-	}
+	return null;
 }
 
-export default connect(
-	( state, { query, siteId } ) => ( {
-		requestingSettings: isRequestingJetpackSettings( state, siteId, query ),
-	} ),
-	{ requestJetpackSettings }
-)( QueryJetpackSettings );
+QueryJetpackSettings.propTypes = {
+	siteId: PropTypes.number.isRequired,
+	query: PropTypes.object,
```

Comment:
Please note that `query` isn't used by any of the components which render `<QueryJetpackSettings />`, shall we remove it?

Reply:
This was used for the Calypso version of Jetpack Onboarding (see https://github.com/Automattic/wp-calypso/blob/7f01c5fb204799954c9f1f7482876da120a0c44f/client/jetpack-onboarding/main.jsx#L169).

The entire feature was removed long ago in #38000.

I'm very doubtful this will be used in the future, so I'd suggest removing it completely from the query component and the related Redux action creator / reducer / selectors. 

Is this something you'd like to clean up in this PR?

Reply:
> Is this something you'd like to clean up in this PR?

Yup 👍🏻

Reply:
Added https://github.com/Automattic/wp-calypso/pull/58111/commits/f3854235d3ecfb8881b43fc3ef91dd020d8e8556